cmd/juju/application/deployer/bundlehandler.go

```diff
@@ -975,9 +1008,9 @@ func (h *bundleHandler) upgradeCharm(change *bundlechanges.UpgradeCharmChange) e
 		return errors.Errorf("unexpected upgrade charm URL %q", p.Charm)
 	}
 
-	chID := charmstore.CharmID{
-		URL:     cURL,
-		Channel: csparams.Channel(h.origins[*cURL].Risk),
+	chID := application.CharmID{
+		URL:    cURL,
+		Origin: h.origins[*cURL],
```

Comment:
should we use this origin, or the one from resolve charm url above (1004).    It's very confusing for the origins, to have 2 versions of resolveCharmURL, one that returns the origin, a local one that does not.  Can we get rid of the local one to reduce confusion in the future, or at least make it return an origin?

Reply:
If anything we should just always use `h.origins[*cURL]`, at least that way we can always know it should be valid for all requests.

Reply:
Let's consolidate then.  I keep thinking we're going to end up with subtle bugs with so many places to make origins.  

the resolve charm url at (1004) makes me wonder if we're already been through resolveCharmsAndEndpoints().